### PR TITLE
Add playbooks to swap FSS with NFS

### PR DESCRIPTION
**Pull request by AndriyNovykov**
Create 5 playbooks. The swap playbooks swap the FSS home and data directories with the high availability NFS home and data directories. The ha_nfs playbooks are the updated versions of the site, new_nodes, and resize_add playbooks.

### CODE REVIEW

File: playbooks/ha_nfs_new_nodes.yml [hunk 1]
```diff
@@ -0,0 +1,212 @@
+
```
Comment (xksteven) [inline playbooks/ha_nfs_new_nodes.yml:1]: which file calls these? 
Are the missing the starting triple dash?

File: playbooks/ha_nfs_new_nodes.yml [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    - etc-hosts
+      
+- hosts: nfs
+  become: true
+  vars:
+     export_path: "{{ scratch_nfs_path }}"
+     export_name: "scratch"
+     local_path:  "/mnt/localdisk/nfs"
+     iscsi_ip: "{{ cluster_mount_ip }}"
+
```
Comment (xksteven) [inline playbooks/ha_nfs_new_nodes.yml:67]: extra newline

File: playbooks/ha_nfs_new_nodes.yml [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    - include_role: 
+        name: iscsi
+      when: scratch_nfs_type == "block" and scratch_nfs|default(true)|bool
+    - include_role: 
+        name: nfs-server
+      when: scratch_nfs|default(true)|bool
+
+- hosts: compute
+  become: true
+
```
Comment (xksteven) [inline playbooks/ha_nfs_new_nodes.yml:78]: Run a linter.

File: playbooks/ha_nfs_new_nodes.yml [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+    - include_role:
+        name: sssd
+      when: ldap|default(true)|bool
+
+- hosts: compute
+  become: true
+  gather_facts: true
+  tasks: 
+    - include_role: 
+        name: hyperthreading
```
Comment (xksteven) [inline playbooks/ha_nfs_new_nodes.yml:115]: Why are we doing this?

File: playbooks/ha_nfs_site.yml [hunk 5]
```diff
@@ -0,0 +1,340 @@
+
```
Comment (xksteven) [inline playbooks/ha_nfs_site.yml:1]: Missing the three dashes

File: playbooks/ha_nfs_resize_add.yml [hunk 6]
```diff
@@ -0,0 +1,213 @@
+- hosts: compute_to_add
```
Comment (xksteven) [inline playbooks/ha_nfs_resize_add.yml:1]: Missing the three dashes.

Comment (xksteven) [reply playbooks/ha_nfs_resize_add.yml:1]: Is `compute_to_add` a valid host?

File: playbooks/ha_nfs_site.yml [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+# configure if instance_principal is False
+- hosts: bastion
+  become: true
+  tasks:
+    - include_role:
+        name: no_instance_principal
+      when: not inst_prin|bool 
+      
+
```
Comment (xksteven) [inline playbooks/ha_nfs_site.yml:129]: formatting. Run a linter ples.

File: playbooks/swap_data.yml [hunk 8]
```diff
@@ -0,0 +1,48 @@
+---
```
Comment (xksteven) [inline playbooks/swap_data.yml:1]: These should only be run once right?  Need to put in guards in place for that.  `run_once` in ansible or something.

File: playbooks/swap_home.yml [hunk 9]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (xksteven) [inline playbooks/swap_home.yml:1]: Same here about adding guards.

**Review by xksteven — CHANGES_REQUESTED** [review]
Some changes to be made

**Comment by xksteven** [comment]
Add a README to the directory to at least explain your additional playbooks.  We can add in more details later.

File: playbooks/ha_nfs_new_nodes.yml (hunk 1 above)
Comment (AndriyNovykov) [reply playbooks/ha_nfs_new_nodes.yml:1]: The original file didn't have them so I don't believe them.

File: playbooks/ha_nfs_new_nodes.yml (hunk 4 above)
Comment (AndriyNovykov) [reply playbooks/ha_nfs_new_nodes.yml:115]: This was done in the original playbook.

File: playbooks/swap_data.yml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+- hosts: bastion
+  gather_facts: true
+  become: true
+  tasks:
+    - name: Unmount NFS home export
+      mount:
+        path: "/mnt/ha-nfs/home"
+        src: "172.16.4.200:/mnt/nfsshare/exports/home"
+        fstype: nfs
+        state: unmounted
```
Comment (xksteven) [inline playbooks/swap_data.yml:13]: Add when conditonal to check if path exists

File: playbooks/swap_data.yml (hunk 8 above)
Comment (AndriyNovykov) [reply playbooks/swap_data.yml:1]: Running multiple times does not cause any issues. Ansible first checks whether the export is mount or not before trying to mount or unmount.

File: playbooks/ha_nfs_new_nodes.yml (hunk 4 above)
Comment (xksteven) [reply playbooks/ha_nfs_new_nodes.yml:115]: Let's just make a new git issue to remove them

File: playbooks/swap_data.yml (hunk 8 above)
Comment (xksteven) [reply playbooks/swap_data.yml:1]: Have you run it more than once as a test? 
Can you show the output?

In my head I think it'll error out.

File: playbooks/swap_data.yml (hunk 10 above)
Comment (AndriyNovykov) [reply playbooks/swap_data.yml:11]: Removed src and fstype. They are no longer needed for the Ansible mount module. 
https://github.com/ansible/ansible-modules-core/pull/1357 for reference.

File: playbooks/ha_nfs_site.yml [hunk 11]
```diff
@@ -0,0 +1,334 @@
...
+      when: monitoring|default(false)|bool
+    - include_role: 
+        name: grafana
+      when: monitoring|default(false)|bool
+    - include_role: 
+        name: autoscaling_mon
+      when: autoscaling_monitoring|default(false)|bool
+    - include_role: 
+        name: cron
+
```
Comment (xksteven) [inline playbooks/ha_nfs_site.yml:294]: How did this get past the linter?

**Review by xksteven — CHANGES_REQUESTED** [review]
Few more changes.

File: playbooks/README.md [hunk 12]
```diff
@@ -0,0 +1,2 @@
+After the high avilability NFS is deployed rename site.yml, new_nodes.yml, and resize_add.yml by prefixing 'old_' to each of them. 
+Then remove the 'ha_nfs_' prefix from the playbooks that have the 'ha_nfs_'.
```
Comment (xksteven) [inline playbooks/README.md:2]: This is very unclear especially as a README. 

These steps should just already be in a playbook too. 

Stuff I'd expect to see instead (Can include one or more of the following):

- General descriptive of the playbooks/folder
- How to run a playbook
- What happens when you run one (any) of them?
- When should you run one of them?
- Repeat the above two for each playbook.
- Credits feel free to say you wrote some of it.
- maybe give the README a cool title

**Review by xksteven — CHANGES_REQUESTED** [review]
Update readme

File: playbooks/swap_data.yml (hunk 10 above)
Comment (xksteven) [reply playbooks/swap_data.yml:11]: Double check which ansible version were using too to confirm it's okay to remove.

**Comment by xksteven** [comment]
Now that we know it'll come in handy let's add the revert of mounting NFS with FSS

File: playbooks/README.md [hunk 13]
```diff
@@ -0,0 +1,11 @@
+This is the '/playbooks' directory. In this directory are a collection of playbooks that are used to configure the compute cluster and high availability NFS. 
+The original playbooks were written by Oracle Cloud Infrastructure engineers. New playbooks and updates to existing playbooks is done by engineers at the Center for AI Safety. 
+
+Most of these playbooks should not be executed manually. 
+
+Those playbooks that can be run manually are:
```
[the remark this answers is not in the document]
Comment (steven-basart) [reply playbooks/README.md:6]: Example 
- add_root_ssh_key.yml
    - This is run to Add an ssh key to /root to be able to unmount the /home directories. Should only be run once when creating the NFS.  
    
As an aside
When is they key removed?

**Comment by steven-basart** [comment]
I think we can close this now? @andriy-safe-ai

**Comment by steven-basart** [comment]
I think we can safely close this PR at this point. We can maybe revisit the NFS later but several other work needs to be done first. 

Getting example workloads, RCA, and researching the right solution for next time.

**Pull request closed without merging**